Codecademy/gamut
# chore(Button): updating IconButton's variants' icon sizes

### Overview
IconButtons use icons that currently size depending on the parent container using `"calc(100% - 14)"`
But for some reason this calculation seems to be done using the width/height of `::after` which can be marginally larger (e.g. 40.8 when it should be exactly 40)
![image](https://github.com/Codecademy/gamut/assets/25854220/9bd873ec-cf4d-4c6a-adf5-bf98ca6b1d4e)
^ on the prod version of storybook, it comes out to 40 🤷 

Regardless, the calculation of `-14px` should actually be `-18px` to enable a padding of 8px all around.

To make the sizing more explicit, concrete values were used:
normal: 24, <- from figma
small: 16, <- from figma
large: 40, <- assuming since currently a large IconButton is 56 and 56 - 16 = 40, the figma will need to be updated to reflect this size. Confirmed with Stacey

### PR Checklist

- [ ] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=1106-90&m=dev
- [ ] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-702 

#### Testing Instructions

1. Go to https://666b16bc273aa9527803b3d5--gamut-preview.netlify.app/?path=/docs/atoms-button--icon-button 
2. Inspect the IconButton element's variants, check that:
  a. normal: parent container is 40x40 and the icon is 24x24
  b. small: parent container is 32x32 and the icon is 16x16
  c. large: parent container is 56x56 and the icon is 40x40
---
3. Go to the Monolith Preview and `/admin/concessions`
4. Check that edit icon buttons (they're small) and should render correctly (32x32 for parent, 16x16 for the icon)
--- 
5. Go to Mono Preview (Do NOT login)
6. This main page should be the signup page with a carousel and a pause button, inspect the pause button
![image](https://github.com/Codecademy/gamut/assets/25854220/37fe29c2-0c80-468f-b502-1049ae4cd45d)
7. This is also a small variant and should render correctly (32x32 for parent, 16x16 for the icon)
8. ...
9. Profit!


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37875)  | [Monolith Env](https://pr-37875-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/6644)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-6644)   |
